bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/ChannelUpdaterJob.java

```diff
@@ -0,0 +1,285 @@
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.scheduler.SchedulerRunnable;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ChannelUpdaterJob} updates all channel values
+ *
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class ChannelUpdaterJob implements SchedulerRunnable, Runnable {
+
+    private final Thing thing;
+    private final LuxtronikHeatpumpConfiguration config;
+    private final LuxtronikTranslationProvider translationProvider;
+    private final Logger logger = LoggerFactory.getLogger(ChannelUpdaterJob.class);
+
+    public ChannelUpdaterJob(Thing thing, LuxtronikTranslationProvider translationProvider) {
+        this.thing = thing;
+        this.translationProvider = translationProvider;
+        this.config = thing.getConfiguration().as(LuxtronikHeatpumpConfiguration.class);
+    }
+
+    public Thing getThing() {
+        return thing;
+    }
+
+    @Override
+    public void run() {
+        // connect to heatpump and check if values can be fetched
+        final HeatpumpConnector connector = new HeatpumpConnector(config.ipAddress, config.port);
+        final LuxtronikHeatpumpHandler handler = (LuxtronikHeatpumpHandler) thing.getHandler();
+
+        try {
+            connector.read();
+        } catch (IOException e) {
+            logger.warn("Could not connect to heatpump (uuid={}, ip={}, port={}): {}", thing.getUID(), config.ipAddress,
+                    config.port, e.getMessage());
+
+            if (handler != null) {
+                handler.setStatusConnectionError();
+            }
+            return;
+        }
+
+        if (handler != null) {
+            handler.setStatusOnline();
+        }
+
+        // read all available values
+        Integer[] heatpumpValues = connector.getValues();
+
+        // read all parameters
+        Integer[] heatpumpParams = connector.getParams();
+        Integer[] heatpumpVisibilities = connector.getVisibilities();
+
+        for (HeatpumpChannel channel : HeatpumpChannel.values()) {
+            try {
+                Integer rawValue = getChannelValue(channel, heatpumpValues, heatpumpParams, heatpumpVisibilities);
+
+                if (rawValue == null) {
+                    continue;
+                }
+
+                State value = convertValueToState(rawValue, channel.getItemClass(), channel.getUnit());
+
+                if (value != null) {
+                    handleEventType(value, channel);
+                }
+
+            } catch (Exception e) {
+                logger.warn("An error occurred while updating the channel {}: {}", channel.getCommand(),
+                        e.getMessage());
+            }
```

Comment:
can you be more specific here?

Reply:
actually there should no exception be thrown. At least it didn't while testing. It's catched nevertheless so following channels will be updated even if one fails. Will add that as a comment in the code so it clear.